Gems/EMotionFX/Code/EMotionFX/Source/Event.h

```diff
@@ -42,7 +45,21 @@ namespace EMotionFX
         void SetEventData(size_t index, EventDataPtr&& newData);
         void InsertEventData(size_t index, EventDataPtr&& newData);
 
+        void SetEventDataChangeEvent(Event::EventDataChangeEvent::Handler& handler);
     protected:
+        struct EventContainer {
```

Comment:
Why are you copying an event? I think AZ::event aren't supposed to be copied by design. EventHanderA is suppose to only handle EventA. If there is an EventB that copied from EventA, it will trigger undefined behavior.

Reply:
I'm not. The events are discarded when the emfx event is copied. It's done in a couple places.